File: Algorithms/Unionoflists/algorithm1.cpp
#include <bits/stdc++.h>
#include <time.h>
using namespace std;
int main()
{
  srand(time(0));
  int l1=rand()%10;
  int l2=rand()%10;
  int A[l1],B[l2];
  for(int i=0;i<l1;i++)
    A[i]=rand()%10-rand()%10;
  for(int i=0;i<l2;i++)
    B[i]=rand()%10-rand()%10;
  for(int i=0;i<l1;i++)
    cout<<A[i]<<',';
  cout<<"\b \n";
  for(int i=0;i<l2;i++)
    cout<<B[i]<<',';
  cout<<"\b \n";
  int temp=0;
//Algorithm 1: create a set for A and iterate elements of B through it O(nlogn)
  set<int> s;
  for(auto x:A)
    s.insert(x);
  set<int> unionAB;
  for(auto x:B)
    if(s.count(x))
    {
      temp++;
      unionAB.insert(x);
    }
  cout<<"Number of common elements=" <<temp;
}

File: Algorithms/Unionoflists/algorithm1.py
import random
A=random.sample(range(-10,10),random.randint(0,10))
B=random.sample(range(-10,10),random.randint(0,10))
print(A)
print(B)
temp=0
#Algorithm 1: create a set for A and iterate elements of B through it O(nlogn)
s=set()
unionAB=[]
for x in A:
    s.add(x)
for x in B:
    if x in s:
        temp=temp+1
        unionAB.append(x)
print(unionAB)
